wandb/sdk/artifacts/artifact.py

```diff
@@ -1894,6 +1962,10 @@ def link(self, target_path: str, aliases: Optional[List[str]] = None) -> None:
         """
         if self._state == ArtifactState.PENDING:
             raise ArtifactNotLoggedError(self, "link")
+        if self._ttl_changed:
```

Comment:
We want to show the warning if the artifact has a TTL rule, i.e.:
```suggestion
        if not self._ttl_is_inherited:
```